### PR TITLE
Add user agent

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: multiversx_sdk/network_providers/user_agent.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+from multiversx_sdk.network_providers.config import NetworkProviderConfig
+from multiversx_sdk.network_providers.constants import UNKNOWN_CLIENT_NAME
+
+logger = logging.getLogger("user_agent")
+
+
+def extend_user_agent(user_agent_prefix: str, config: NetworkProviderConfig):
+    if config.client_name is None:
+        logger.warning("Please provide the client name of the application that uses the SDK. It will be used for metrics.")
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/user_agent.py:12]: Should be info, similar to JS.

_We recommend providing the name of your application when instantiating a NetworkProvider (e.g., ProxyNetworkProvider, ApiNetworkProvider). This information will be used for metrics collection and improving our services._

Comment (popenta) [reply multiversx_sdk/network_providers/user_agent.py]: Changed the message to the one you've mentioned

File: multiversx_sdk/network_providers/user_agent.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 def extend_user_agent(user_agent_prefix: str, config: NetworkProviderConfig):
     if config.client_name is None:
-        logger.warning("Please provide the client name of the application that uses the SDK. It will be used for metrics.")
+        logger.warning("We recommend providing the `client_name` when instantiating a NetworkProvider (e.g. ProxyNetworkProvider, ApiNetworkProvider). This information will be used for metrics collection and improving our services.")
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/user_agent.py:12]: Also, not `warning`, but `info` instead.

Comment (popenta) [reply multiversx_sdk/network_providers/user_agent.py]: Done

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**